nanshe_ipython.ipynb

```diff
@@ -1147,8 +1148,14 @@
         "    plot_projs.append(plot_std)\n",
         "\n",
         "\n",
+        "all_tr_dtype_srcs = ColumnDataSource(data=dict(traces_dtype=traces.dtype.type(0)[None]))\n",
         "all_tr_shape_srcs = ColumnDataSource(data=dict(traces_shape=traces.shape))\n",
-        "all_tr_srcs = ColumnDataSource(data=dict(traces=traces.flatten()))\n",
+        "all_tr_srcs = ColumnDataSource(data=dict(\n",
+        "    traces=numpy.frombuffer(\n",
+        "        zlib.compress(traces.tobytes()),\n",
```

Comment:
First compress the `byte`-string representation of the traces using `zlib`. Then store the result back as a NumPy `uint8` array.

This last step leverages Bokeh 0.12.4's `base64` encoding of NumPy arrays to compactly store the result. This also guarantees that BokehJS will decode this to a `Uint8Array` in JavaScript, which we can easily pass to pako for decompression.